Module-2/README.md

```diff
@@ -0,0 +1,175 @@
+
+Continuous Learning is the process of consistently updating and expanding one's knowledge and skills in a specific area.
+
+Continuous learning is crucial in software development, much like in the field of medicine, where there is a vast and ever-evolving body of knowledge. Engaging in continuous learning ensures that developers remain relevant and informed about the latest technologies and practices and retain existing skills.
+
+#### _c) How can you find out what modifiers a View has?_
+
+There are a number of ways to find modifiers for a View, including
+
+- Checking the Developer Documentation for the specific View.
+- Referring to `SwiftUI` > `Views` > `View fundamentals` >  `Configuring view elements` subsections in Developer Documentation.
+- Select the View in the editor window or the Canvas (when in `Selectable` mode), then go to the `Attributes Inspector` in the `Inspectors` side drawer.  
+- Use the `Modifiers` section of the `Library` and scroll to the section relevant for the View (eg. Text).
+
+#### _d) What is a breakpoint?_
+
+A breakpoint is a way to interrupt and pause running code at a defined line in order to debug and inspect variables in the app.  The execution of code can then be controlled and stepped through.
+
+#### _e) How can you access environment values in your App?_
+
+With the `@Environment` property wrapper using the associated `EnvironmentValue` key path for the relevant property.
+
+eg. 
+```
+    @Environment(\.verticalSizeClass) var verticalSizeClass
+```
+
+#### _f) How can you determine, in code, if the App is in Dark or Light Mode?_
+
+By checking the `colorScheme` environment value:
+
+```
+@Environment(\.colorScheme) var colorScheme
+```
+
+Which will be either: `ColorScheme.light` or `ColorScheme.dark`
+
+#### _g) Why are magic numbers an issue, and how should you avoid them?_
+
+“Magic numbers” are hard-coded numeric values that are used directly in code  The term “magic” is used because the number has unexplained meaning.
+
+Magic numbers issues:
+- **Reduced readability** of the code (hard to understand what the number represents without context).
+-  **Reduced maintainability** (if the same number is used in multiple places, it needs to be changed everywhere it occurs; are all cases where that number is used representing the same concept?).
+- **Difficulty debugging** (if the magic number is causing the issue, where did it come from, and why is it there?).
+
+Magic numbers can be avoided by using named constants or creating sets of related constants (e.g., using enums or static variables in structs).
+
+#### _h) How can you view your App in Light and Dark Modes simultaneously?_
+
+- Choosing the `Color Scheme Variants` in the Xcode Canvas.
+- Creating multiple previews in the `PreviewProvider` for the `View` with associated `.preferredColorScheme(_ colorScheme: ColorScheme?) ` modifiers and switch between the two.
+
```

Comment:
Just a note here if someone picky asks ;)
In Swift by convention, the documentation is always referring to methods by omitting parameter names like `colorScheme` and types like `ColorScheme`. So we would write it like this `preferredColorScheme(_:)`.

Reply:
Noted.  Will do in the future.  Thanks.